### PR TITLE
MacSymbolicator: submission

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by i0ntempest** [comment]
```
  Failed to determine if database is empty or not: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “org.swift.swiftpm” in the folder “Caches”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSUnderlyingError=0x600003838810 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}Failed to determine if database is empty or not: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “org.swift.swiftpm” in the folder “Caches”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSUnderlyingError=0x600003962610 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “repositories” in the folder “SourcePackages”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Developer/Xcode/DerivedData/MacSymbolicator-ciurkbilovnqvvazcnoxwgejefal/SourcePackages/repositories, NSUnderlyingError=0x6000038392f0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
  2024-06-11 06:34:20.519 xcodebuild[5886:33211] Writing error result bundle to /var/folders/33/2j_9q6d13j7691ws8c74kt940000gp/T/ResultBundle_2024-11-06_06-34-0020.xcresult
  xcodebuild: error: Could not resolve package dependencies:
    Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “repositories” in the folder “SourcePackages”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Developer/Xcode/DerivedData/MacSymbolicator-ciurkbilovnqvvazcnoxwgejefal/SourcePackages/repositories, NSUnderlyingError=0x6000038392f0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
  
  Command failed: cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_MacSymbolicator/MacSymbolicator/work/MacSymbolicator-2.6" && /usr/bin/xcodebuild  -target "MacSymbolicator" -configuration Release build INSTALL_PATH=/Applications/MacPorts  OBJROOT="/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_MacSymbolicator/MacSymbolicator/work/MacSymbolicator-2.6/build/" SYMROOT="/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_devel_MacSymbolicator/MacSymbolicator/work/MacSymbolicator-2.6/build/" MACOSX_DEPLOYMENT_TARGET=14.0 ARCHS=arm64 SDKROOT="/Applications/Xcode_15.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.sdk" GCC_VERSION=com.apple.compilers.llvm.clang.1_0 CLANG_CXX_LIBRARY="libc++" CODE_SIGN_IDENTITY=-
  Exit code: 74
  Error: Failed to build MacSymbolicator: command execution failed
  DEBUG: Error code: NONE
  DEBUG: Backtrace: command execution failed
      while executing
  "$procedure $targetname"
  Error: See /opt/local/var/macports/logs/_Users_runner_work_macports-ports_macports-ports_ports_devel_MacSymbolicator/MacSymbolicator/main.log for details.
  Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
  Error: Processing of port MacSymbolicator failed
  Build of 'MacSymbolicator' failed.
```
Is this something wrong in the MP base? Or maybe Xcode PG?

**Comment by ryandesign** [comment]
Sounds like https://trac.macports.org/ticket/57137

File: devel/MacSymbolicator/Portfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    xcode.build.settings-append \
+        CODE_SIGN_IDENTITY=-
+} else {
+    xcode.build.settings-append \
+        CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO
+}
+
+destroot {
+    file copy ${worksrcpath}/build/${xcode.configuration}/${name}.app ${destroot}${applications_dir}
+    ln -s ${destroot}${applications_dir}/${name}.app/Contents/MacOS/MacSymbolicatorCLI ${prefix}/bin/macsymbolicator-cli
```
Comment (ryandesign) [inline devel/MacSymbolicator/Portfile:36]:
```suggestion
    ln -s ${applications_dir}/${name}.app/Contents/MacOS/MacSymbolicatorCLI ${destroot}${prefix}/bin/macsymbolicator-cli
```

File: devel/MacSymbolicator/Portfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+github.setup        inket MacSymbolicator 2.6
+revision            0
+categories          devel aqua
+maintainers         {i0ntempest @i0ntempest} openmaintainer
+license             GPL-2
+
+description         Symbolicate macOS/iOS crash reports easily
+long_description    A simple Mac app for symbolicating macOS/iOS crash reports, supports symbolicating\
+                    .crash and .ips crash reports, and also sample and spindump reports
```
Comment (ryandesign) [inline devel/MacSymbolicator/Portfile:15]:
```suggestion
long_description    A simple Mac app for symbolicating macOS/iOS crash reports. \
                    supports symbolicating .crash and .ips crash reports, and \
                    also sample and spindump reports.
```

File: devel/MacSymbolicator/Portfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+checksums           rmd160  b2e0859c6a773f2c019aed0c038f02d8a8718365 \
+                    sha256  2d75ae08bbb0892dce28fe14a333736687d6c8fb01c1e4382524df099c60921b \
+                    size    1265658
+
+platforms           macosx
+
```
Comment (ryandesign) [inline devel/MacSymbolicator/Portfile:22]: The xcode portgroup sets this already.
```suggestion
```

**Comment by i0ntempest** [comment]
Might be something else, or at least the old workaround no longer works. The permission error is under `${prefix}/var/macports/home/Library/Caches/org.swift.swiftpm/`, so something in the macports user home.

**Comment by markemer** [comment]
I'm working on an Xcode 1.1 PG to deal with workspacee, derived data, and a possible need to run swift package manager under `xcodebuild` but that may be a while.

That last thing might actually be the problem, seems like macOS thinks that running the equivalent xcodebuild to `swift package` is trying to escape the sandbox, why that is? I'm not sure, but it must be trying to fetch the git clones to someplace we don't expect:
```
   Resolve Package Graph
  Failed to determine if database is empty or not: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “org.swift.swiftpm” in the folder “Caches”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSUnderlyingError=0x6000006ed5f0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
  Fetching from https://github.com/apple/swift-argument-parser.git
  
  Invalid manifest
  sandbox-exec: sandbox_apply: Operation not permitted in https://github.com/apple/swift-argument-parser.git2024-06-15 02:07:52.306 xcodebuild[4985:21970] Writing error result bundle to /var/folders/kn/x_bx9qmn4tx5k3hx7s68cs080000gp/T/ResultBundle_2024-15-06_02-07-0052.xcresult
  xcodebuild: error: Could not resolve package dependencies:
    Invalid manifest
  sandbox-exec: sandbox_apply: Operation not permitted in https://github.com/apple/swift-argument-parser.git
```

**Comment by i0ntempest** [comment]
@markemer I have another draft at https://github.com/macports/macports-ports/pull/25831 which is macOS 15 only. The project also uses SPM and it's failing in a similar way (permission error), though sandbox isn't mentioned. That path mentioned in the log should be writable by the `macports` user but Xcode claims it's not...
```
:info:build Resolve Package Graph
:info:build /opt/local/var/macports/home/Library/org.swift.swiftpm/configuration is not accessible or not writable, disabling user-level cache features./opt/local/var/macports/home/Library/org.swift.swiftpm/security is not accessible or not writable, disabling user-level cache features./opt/local/var/macports/home/Library/Caches/org.swift.swiftpm is not accessible or not writable, disabling user-level cache features.Failed to determine if database is empty or not: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “org.swift.swiftpm” in the folder “Caches”." UserInfo={NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSUnderlyingError=0x600002456190 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}Failed to determine if database is empty or not: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “org.swift.swiftpm” in the folder “Caches”." UserInfo={NSUnderlyingError=0x600002496190 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}, NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm, NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm}
:info:build Fetching from https://github.com/apple/swift-argument-parser.git
:info:build failed loading cached manifest for 'swift-argument-parser': Error Domain=NSCocoaErrorDomain Code=4 "The file “manifests” doesn’t exist." UserInfo={NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x60000256c000 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}, NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests}failed closing manifest db cache: Error Domain=NSCocoaErrorDomain Code=4 "The file “manifests” doesn’t exist." UserInfo={NSUnderlyingError=0x600002750e40 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}, NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests, NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests}Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “ManifestLoading” in the folder “manifests”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests/ManifestLoading, NSUnderlyingError=0x6000027510e0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}, NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests/ManifestLoading} in https://github.com/apple/swift-argument-parser.git2024-09-19 05:49:25.457 xcodebuild[53826:7640424] Writing error result bundle to /opt/local/var/macports/build/_Users_Admin_Developer_Ports_sysutil_eligibility-util/eligibility-util/work/.tmp/ResultBundle_2024-19-09_05-49-0025.xcresult
:info:build xcodebuild: error: Could not resolve package dependencies:
:info:build   Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “ManifestLoading” in the folder “manifests”." UserInfo={NSFilePath=/opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests/ManifestLoading, NSUnderlyingError=0x6000027510e0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}, NSURL=file:///opt/local/var/macports/home/Library/Caches/org.swift.swiftpm/manifests/ManifestLoading} in https://github.com/apple/swift-argument-parser.git
:info:build Command failed: cd "/opt/local/var/macports/build/_Users_Admin_Developer_Ports_sysutil_eligibility-util/eligibility-util/work/eligibility-0.2.1-beta" && /usr/bin/xcodebuild -derivedDataPath ./DerivedData -project "eligibility.xcodeproj" -scheme "eligibility_util" -configuration Release build INSTALL_PATH=/Applications/MacPorts_  OBJROOT="/opt/local/var/macports/build/_Users_Admin_Developer_Ports_sysutil_eligibility-util/eligibility-util/work/eligibility-0.2.1-beta/build/" SYMROOT="/opt/local/var/macports/build/_Users_Admin_Developer_Ports_sysutil_eligibility-util/eligibility-util/work/eligibility-0.2.1-beta/build/" MACOSX_DEPLOYMENT_TARGET=15.0 ARCHS=arm64 SDKROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.sdk" GCC_VERSION=com.apple.compilers.llvm.clang.1_0 CLANG_CXX_LIBRARY="libc++" CODE_SIGN_IDENTITY=- OTHER_SWIFT_FLAGS=--disable-sandbox
:info:build Exit code: 74
```

**Comment by reneeotten** [comment]
No progress here for quite some time now. Feel free to reopen once the issues are resolved and it's ready to merge.

**Comment by i0ntempest** [comment]
Still waiting for the new Xcode/SwiftPM pg to resolve whatever this is. I also have another PR blocked by the same issue.

**Pull request closed without merging**